spec/orchestration_executor_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+RSpec.describe Simplekiq::OrchestrationExecutor do
+  let(:workflow) do
+    [
+      {"klass" => "OrcTest::JobA", "args" => [1]}
+    ]
+  end
+
+  let(:classname) { "FakeOrchestration" }
+
+  describe ".execute" do
```

Comment:
I feel like there should be a test that asserts a batch is returned from `OrchestrationJob.perform`

Reply:
@austenmadden There wouldn't be, `OrchestrationJob.perform` is only ever called by `Sidekiq` internals so whether or not it returns a batch is irrelevant as nothing in the `Sidekiq` code would be consuming that. Maybe you meant `perform_async`? But again, that would only return a job id not a batch id, which is dictated by internal `Sidekiq` behavior.

Maybe you meant that the orchestration batch is accessible from `perform_orchestration`? But it still wouldn't be, because `Simplekiq` first runs `perform_orchestration` to generate the workflow structure, then starts actually creating batches and spinning off steps afterwards.

So what you'd want to do for your specific purposes of wanting to track that orchestration batch is probably add a job as the first step of the orchestration which looks at its batch's parent batch (it's batch = the step batch, the step batch's parent = the orchestration batch)

Reply:
🤔 Ok; I think that was part of my confusion. Originally when approaching this, I believed it was possible to have the entry point to the orchestration expose the batch as a return value rather than a job id but I can see how returning a job id would be inline with how `perform_async` and friends works. Misunderstanding of the change on my part.

Reply:
We discussed offline and the store-it-in-the-first-step approach is likely what he's going to go with. An alternative that came up in case others are reading along is one could open a batch around the initial queue of the orchestration job, like so:

```ruby
batch = Sidekiq::Batch.new
batch.jobs do
  WhateverOrchestrationJob.perform_async(args)
end
return batch.bid
```

Something like that I think. The downside with this approach is it only works if you queue the orchestration like this. If you make the orchestration a step in some other orchestration then it would be more complicated, and not really generalizable. The nice thing about doing it in the first job is that it should Just Work no matter how the orchestration gets queued, even if it has parent batches/orchestrations it'll still be able to trivially find the batch at the head of the orchestration in question.